catalyst-gateway/bin/src/cardano/mod.rs

```diff
@@ -130,159 +132,130 @@ async fn process_blocks(
     follower: &mut Follower, db: Arc<EventDB>, network: Network, machine_id: MachineId,
     genesis_values: &GenesisValues,
 ) {
-    loop {
-        let chain_update = match follower.next().await {
-            Ok(chain_update) => chain_update,
-            Err(err) => {
-                error!(
-                    "Unable to receive next update from the {network:?} follower err: {err} - skip..",
-                );
-                continue;
-            },
-        };
+    info!("Follower started processing blocks");
+
+    let mut blocks_buffer = Vec::new();
 
-        match chain_update {
-            ChainUpdate::Block(data) => {
-                let block = match data.decode() {
-                    Ok(block) => block,
-                    Err(err) => {
-                        error!("Unable to decode {network:?} block {err} - skip..",);
-                        continue;
+    let mut ticker = tokio::time::interval(Duration::from_secs(1));
```

Comment:
I totally understood what the intention of this `ticker`, seems it makes sense not to do it as a part of this PR, and provide some kind of this mechanics as part of this ticket https://github.com/input-output-hk/catalyst-voices/issues/452. Because maybe we want to do it another way, and most probable we want to configure this parameter from configs. And all these changes seems really out of scope of this PR and could just blow up it.
So maybe lets just remove this part for now and later just design the proper mechanics of synchronisation in general.
What do you think ?

Reply:
This logic is necessary for the current implementation of batch writes. Without it, the DB data can become stale until enough blocks are received and the buffer is filled. This ticker is there just to prevent that. We need to set a sensible interval (1 second is too low, I left that by mistake).

That said, this has nothing to do with syncing multiple `catalyst-gateway` instances. This PR only contains logic related to writing chain data to the database.

Reply:
I agree with @FelipeRosa Batching needs a time bounds.  batch this much data I can process in this time and then commit it.
We don't want to be bound by the number of blocks we can get from the blockchain we want to be bound by processing intervals.
If one was worried about memory usage you could put an upper bound on that, but I don't think thats necessary here.  better to just bound on time.